cmd/influxd/inspect/export.go

```diff
@@ -0,0 +1,409 @@
+package inspect
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kit/cli"
+	"github.com/influxdata/influxdb/v2/models"
+	"github.com/influxdata/influxdb/v2/pkg/escape"
+	"github.com/influxdata/influxdb/v2/tsdb/engine/tsm1"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// exportFlags contains CLI-compatible forms of export options.
+type exportFlags struct {
+	enginePath   string
+	bucketID     influxdb.ID
+	measurements []string
+	startTime    string
+	endTime      string
+
+	outputPath string
+	compress   bool
+
+	logLevel zapcore.Level
+}
+
+// exportFilters contains storage-optimized forms of parameters used to restrict exports.
+type exportFilters struct {
+	measurements map[string]struct{}
+	start        int64
+	end          int64
+}
+
+func newFilters() *exportFilters {
+	return &exportFilters{
+		measurements: make(map[string]struct{}),
+		start:        math.MinInt64,
+		end:          math.MaxInt64,
+	}
+}
+
+// filters converts CLI-specified filters into storage-optimized forms.
+func (f *exportFlags) filters() (*exportFilters, error) {
+	filters := newFilters()
+
+	if f.startTime != "" {
+		s, err := time.Parse(time.RFC3339, f.startTime)
+		if err != nil {
+			return nil, err
+		}
+		filters.start = s.UnixNano()
+	}
+
+	if f.endTime != "" {
+		e, err := time.Parse(time.RFC3339, f.endTime)
+		if err != nil {
+			return nil, err
+		}
+		filters.end = e.UnixNano()
+	}
+
+	for _, m := range f.measurements {
+		filters.measurements[m] = struct{}{}
+	}
+
+	return filters, nil
+}
+
+func newFlags() *exportFlags {
+	return &exportFlags{
+		logLevel: zapcore.InfoLevel,
+		compress: false,
+	}
+}
+
+// NewExportCommand builds and registers the `export` subcommand of `influxd inspect`.
+func NewExportCommand(v *viper.Viper) *cobra.Command {
+	flags := newFlags()
+
+	cmd := &cobra.Command{
+		Use:   `export`,
```

Comment:
Is `export` too generic? Should it be `export-tsm`? Or maybe `export-line-protocol`?

Reply:
@timhallinflux any thoughts on this + the options and help text below?

Reply:
proposing `export-lp`